test/ui/test_ui_devices.py

```
@@ -82,29 +77,36 @@ def test_devices_totals_api(api_token):
     driver.get(f"{BASE_URL}/devices.php")
-    time.sleep(2)
+    wait_for_page_load(driver, timeout=10)
 
     # --- Click "Add Device" ---
-    add_buttons = driver.find_elements(By.CSS_SELECTOR, "button#btnAddDevice, button[onclick*='addDevice'], a[href*='deviceDetails.php?mac='], .btn-add-device")
-    if not add_buttons:
+    # Wait for the "New Device" link specifically to ensure it's loaded
+    add_selector = "a[href*='deviceDetails.php?mac=new'], button#btnAddDevice, .btn-add-device"
+    try:
+        add_button = wait_for_element_by_css(driver, add_selector, timeout=10)
+    except Exception:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)